Automattic/newspack-blocks
# feat(modal-checkout): add filter for order button text

### All Submissions:






### Changes proposed in this Pull Request:



Adds a filter to allow customizing the text for the button to complete an order in modal checkout.

### How to test the changes in this Pull Request:

1. Check out this branch.
2. As a reader, start the following transactions via modal checkout (Donate block or Checkout Button block for non-donations):
  - One-time donation
  - Recurring donation
  - One-time non-donation purchase
  - Subscription non-donation purchase

3. Confirm that the order button text displays as expected, unchanged from behavior in `master` (for any Newspack donation products: "Donate now", or whatever the default button label is for non-donation products).
4. Add some custom code to your test site in a temporary plugin or in your theme's functions.php:

```php
/**
 * Customize the order button text.
 *
 * @param string $text The order button text.
 * @param bool   $is_donation True if the current cart contains a Newspack donation product.
 * @param bool   $is_subscription True if the current cart contains a subscription product.
 * @return string The new text.
 */
function newspack_test_order_button_text( $text, $is_donation, $is_subscription ) {
	if ( $is_donation ) {
		$text = $is_subscription ? 'Custom text for recurring donations' : 'Custom text for one-time donations';
	} else {
		$text = $is_subscription ? 'Custom text for non-donation subscriptions' : 'Custom text for one-time non-donation purchases';
	}

	return $text;
}
\add_filter( 'newspack_blocks_modal_checkout_order_button_text', 'newspack_test_order_button_text', 10, 3 );

```

5. Repeat step 2, and confirm that your custom text is shown as expected in each case.

<img width="604" alt="Screenshot 2024-01-23 at 3 06 38 PM" src="https://github.com/Automattic/newspack-blocks/assets/2230142/15f52f02-0d25-4d83-9276-63321b759fac">

### Other information:
